main-unix.c

```diff
@@ -257,14 +293,21 @@ main (int argc,
 
       flags |= JERRY_FLAG_PARSE_ONLY;
 
-      if (save_snapshot_file_name_p == NULL
-          && ++i < argc)
+      if (save_snapshot_file_name_p != NULL)
+      {
+        JERRY_ERROR_MSG ("Error: snapshot file name already specified\n");
+        print_usage (argv);
+        return JERRY_STANDALONE_EXIT_CODE_FAIL;
+      }
+
+      if (++i < argc)
```

Comment:
One more request. Organized like this, we have check#1 -> bail out, check#2 -> do the thing, else -> bail out. I'd suggest (check -> bail out)* and if no checks failed -> do the thing (an then `else` is not needed).

Several occurrences below.